### ISSUE TITLE
Code Generator bundle

### ISSUE THREAD

**Issue by charbull** [komma/komma#11]
The code generator bundle is not generated into a .jar in the features

**Comment by kenwenzel** [comment]
We could add the bundle to the SDK feature. The problem is that we don't have any UI integration to invoke the generator from Eclipse IDE. Hence it would be useful to create a runnable "uber" jar for the generator that includes the required dependencies and can be invoked from the command line.

**Comment by kenwenzel** [comment]
What is your use case? We currently write most of the interfaces by ourselves.

**Comment by charbull** [comment]
Hi @kenwenzel ,
I am trying to understand how Komma works.
In this case, the added value of the Komma ontology editor is the fact that it is integrated in Eclipse. 
So it will enable a `model and code` approach in the same IDE?

Thank you

**Comment by kenwenzel** [comment]
Yes, KOMMA is essentially the same as the Eclipse Modeling Framework EMF but for RDF instead of (E)MOF. KOMMA's OWL editor is also just one example for what can be done on top of the Eclipse Platform. One of the goals of KOMMA is to provide a comparable toolset for RDF as EMF does for object-oriented model driven applications. Unfortunately, IDE support for code generation is not yet implemented.

**Comment by charbull** [comment]
Thank you for the clarification @kenwenzel !

**Comment by kenwenzel** [comment]
I think this should be reopened since UI integration of the generator is on our road map.

**Comment by vorburger** [comment]
This sounds like there is already code generator somewhere, but it's just not exposed via the Eclipse UI, and/or bundled for headless distribution... but https://komma.enilink.net/docs/ doesn't seem to mention codegen anywherem? Would anyone reading this be able to shed some light on what this is about, and where existing related code, if any, is?

**Comment by vorburger** [comment]
https://github.com/komma/komma/tree/master/bundles/sdk/net.enilink.komma.generator/src/net/enilink/komma/generator ?

**Comment by kenwenzel** [comment]
@vorburger Yes, this is the code generator. Most code is forked from Alibaba https://bitbucket.org/openrdf/alibaba/src/master/
We haven't used it much so it's not well maintained.
